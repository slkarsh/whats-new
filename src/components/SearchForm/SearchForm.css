header {
    display: flex;
    flex-direction: row;
    background-color: white;
    width: 74%;
    position: absolute;
    height: 80px;
    float: right;
    margin-left: 26%;
    border: 2px solid #dedcde;
}

input {
    width: 84%;
    font-size: 1.4em;
}

button {
    width: 15%;
    font-size: 0.95em;
}