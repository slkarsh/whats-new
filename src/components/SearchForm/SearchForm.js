import React, { Component } from 'react';
import './SearchForm.css';

// SEARCHFORM COMPONENT CODE GOES HERE

class SearchForm extends Component {
    constructor() {
        super();
        this.state = {
            input: ''
        }
    }

    handleChange = (event) => {
        this.setState({input: event.target.value.toUpperCase()})
    }

    resetSearchBar = () => {
        this.setState({input: ''})
    }

  searchArticles = (e) => {
    e.preventDefault();
    this.props.searchArticles(this.state.input)
    this.resetSearchBar()
  }

  keyPressed = event => {
      if (event.key === 'Enter') {
          this.searchArticles(event);
      }
  }

    render() {
        return(
            <header>
            <input 
                type='text' 
                placeholder='Search articles'
                name='search'
                value={this.state.input}
                onChange={this.handleChange}
                onKeyPress={this.keyPressed}
            />
            <button onClick={event => this.searchArticles(event)}>Search</button>
        </header>
        )
        
    }
}

export default SearchForm;