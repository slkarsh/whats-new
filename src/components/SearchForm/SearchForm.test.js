import React from 'react'
import { shallow } from 'enzyme'
import SearchForm from './SearchForm'

describe('SearchForm', () => {
    let wrapper
    const mockSearchArticles = jest.fn()

    beforeEach(() => {
        wrapper = shallow(<SearchForm searchArticles={mockSearchArticles} />)
    })

    it('should match the snapshot when rendered correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should update state on handleChange', () => {
        const mockEvent = {target:{ value: 'the who'}}
        const expectedResult = 'THE WHO'

        wrapper.instance().handleChange(mockEvent)

        expect(wrapper.state('input')).toEqual(expectedResult)
    })
})