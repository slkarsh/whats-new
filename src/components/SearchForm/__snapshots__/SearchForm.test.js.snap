// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot when rendered correctly 1`] = `
<header>
  <input
    name="search"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search articles"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</header>
`;
